conn.go

```diff
@@ -882,13 +882,28 @@ func (cn *conn) Exec(query string, args []driver.Value) (res driver.Result, err
 	return r, err
 }
 
-func (cn *conn) send(m *writeBuf) {
+func (cn *conn) send(m *writeBuf) error {
 	_, err := cn.c.Write(m.wrap())
+	return err
+}
+
+func (cn *conn) mustSend(m *writeBuf) {
+	err := cn.send(m)
 	if err != nil {
 		panic(err)
 	}
 }
 
+func (cn *conn) mustSendRetryable(m *writeBuf) {
+	err := cn.send(m)
+	if err != nil {
+		if _, ok := err.(*net.OpError); ok {
```

Comment:
Multiple SQL queries can be passed to at least the `db.Exec()` method like:
`DELETE 1 FROM TABLE; DELETE 2 FROM TABLE;`

It might possible that `cn.send(m)` returns an error after a whole statement was successfully transmitted.
The PostgreSQL server might be implemented in a way that executes the statement already as soon as a termination symbol (`;`) is received.
In this case part of the queries might be rerun if `netErrorNoWrite` is returned for those cases.

To prevent this case we can additionally check if the number of bytes that `cn.c.Write(m.wrap())` returned when an error happens are less then the length up to the first `;` in the statement (or the shortest valid postgreSQL statement). Only in this case we return `netErrorNoWrite`.

To keep it more simple, I would suggest we only return `netErrorNoWrite` if the number of bytes that were successfully written to the socket was `0` and error type is `net.OpError`.

Reply:
I pushed a commit to fix it.